rlclientlib/azure_factories.cc

```diff
@@ -40,41 +47,97 @@ namespace reinforcement_learning {
     return url;
   }
 
+  int observation_api_sender_create(i_sender** retval, const u::configuration& cfg, error_callback_fn* error_cb, i_trace* trace_logger, api_status* status) {
```

Comment:
nit: it seems to be a lot of shared code except OBSERVATION_APIM_TASKS_LIMIT, OBSERVATION_APIM_MAX_HTTP_RETRIES, 
INTERACTION_APIM_TASKS_LIMIT,
INTERACTION_APIM_MAX_HTTP_RETRIES

Reply:
also for the method observation_sender_create, interaction_sender_create

Reply:
Refactored the interaction/observation api sender code I added in this PR.